presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -381,10 +435,14 @@ public void addInput(Page page)
                         expectedGroups,
                         groupByTypes,
                         groupByChannels,
+                        sourceTypes,
+                        orderByChannels,
+                        orderings,
                         hashChannel,
                         operatorContext,
                         maxPartialMemory,
-                        joinCompiler);
+                        joinCompiler,
+                        Optional.of(pageIndexFactory));
```

Comment:
If `spillEnabled` is true (lines 448 - 460 in this class), you should do a `checkState` that `orderings` and `orderByChannels` are empty and document that spilling will crash when used with `ORDER BY` in aggregation.

Furthermore, such `checkstate` should also be done somewhere in the planning phase. Or other way around, this using `ORDER BY` in aggregation should disable spilling.